components/datadog/agent/host.go

```diff
@@ -107,8 +105,17 @@ func (h *HostAgent) installAgent(env *config.CommonEnvironment, params *agentpar
 	}
 
 	// Restart the agent when the HostInstall itself is done, which is normally when all children are done
+	// Behind the scene `DependOn(h)` is transformed into `DependOn(<children>)`, the ComponentResource is skipped in the process.
+	// With resources, Pulumi works in the following order:
+	// Create -> Replace -> Delete.
+	// The `DependOn` order is evaluated separately for each of these phases.
+	// Thus, when an integration is deleted, the `Create` of `restartAgentServices` is done as there's no other `Create` from other resources to wait for.
+	// Then the `Delete` of `restartAgentServices` is done, which is not waiting for the `Delete` of the integration as the dependecy on `Delete` is in reverse order.
 	_, err = h.manager.restartAgentServices(
-		pulumi.Array{configFiles["datadog.yaml"], configFiles["system-probe.yaml"], configFiles["security-agent.yaml"], pulumi.String(intgHash)},
+		func(name string, args command.Args) (string, command.Args) {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
		// use a Customizer function to add triggers to the command
		func(name string, args command.Args) (string, command.Args) {
```